### PR TITLE
Orchestration oop

### PR DESCRIPTION
**Pull request by allegra-noto**

### CODE REVIEW

**Review by Akhan221 — APPROVED** [review]
I already looked over code and gave feedback prior to the PR being created but this looks good to me. I also tried running example notebooks off of this branch and successfully hit the pipeline created endpoint.

File: google_cloud_automlops/deployments/github_actions/builder.py [hunk 1]
```diff
@@ -52,65 +54,23 @@ def build(config: GitHubActionsConfig):
...
-    Returns:
-        str: Contents of github_actions.yaml.
-    """
-    template_file = import_files(GITHUB_ACTIONS_TEMPLATES_PATH) / 'github_actions.yaml.j2'
-    with template_file.open('r', encoding='utf-8') as f:
-        template = Template(f.read())
-        return template.render(
-            artifact_repo_location=artifact_repo_location,
-            artifact_repo_name=artifact_repo_name,
+    write_file(
```
Comment (chasinandrew) [inline google_cloud_automlops/deployments/github_actions/builder.py:57]: With this implementation being so similar to cloud build, I think we can combine the cloudbuild and github_actions deployments into a single file with generic logic to write the jinja file then each deployment type can be a class

File: google_cloud_automlops/orchestration/kfp/KFPComponent.py [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+        self.artifact_repo_name = defaults['gcp']['artifact_repo_name']
+        self.project_id = defaults['gcp']['project_id']
+        self.naming_prefix = defaults['gcp']['naming_prefix']
+
+        # Set and create directory for components if it does not already exist
+        component_dir = BASE_DIR + 'components/' + self.component_spec['name']
+
+        # Build necessary folders
+        # TODO: make this only happen for the first component? or pull into automlops.py
+        make_dirs([
```
Comment (chasinandrew) [inline google_cloud_automlops/orchestration/kfp/KFPComponent.py:93]: This should be one of the very first things that runs when KFP is selected as the orchestration framework. Can a pipeline exist without components? If not, I think it makes sense to keep this here. Also, if other orchestration frameworks require different directory styles, make_dirs function should be moved from utils to a method or util within the KFP classes.

File: google_cloud_automlops/orchestration/kfp/KFPComponent.py [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+            'python3',
+            f'''/pipelines/component/src/{self.component_spec['name']+'.py'}''']
+
+        # Write license and component spec to the appropriate component.yaml file
+        comp_yaml_path = component_dir + '/component.yaml'
+        write_file(
+            filepath=comp_yaml_path,
+            text=GENERATED_LICENSE,
+            mode='w')
+        write_yaml_file(
```
Comment (chasinandrew) [inline google_cloud_automlops/orchestration/kfp/KFPComponent.py:135]: What's the difference between write_file and write_yaml_file? Could there be a type added to the argument of write_file to support yaml? 
Such as: 
```     
write_file(
            filepath=comp_yaml_path,
            text=GENERATED_LICENSE,
            mode='w',
            type='YAML')
```

**Comment by allegra-noto** [comment]
Closing in favor of full OOP PR

**Pull request closed without merging**